### PR TITLE
FMWK-251 Refactor Qualifiers

### PR DESCRIPTION
**Pull request by agrgr**

### CODE REVIEW

File: src/main/java/org/springframework/data/aerospike/core/AerospikeTemplate.java [hunk 1]
```diff
@@ -516,6 +518,21 @@ public <T, S> S findById(Object id, Class<T> entityClass, Class<S> targetClass)
         return (S) findByIdInternal(id, entityClass, targetClass);
     }
 
+    public KeyRecord findByIdInternalWithoutMapping(Object id, Class<?> entityClass, Class<?> targetClass,
```
Comment (roimenashe) [inline src/main/java/org/springframework/data/aerospike/core/AerospikeTemplate.java:521]: If its not part of AerospikeInternalOperations and is internal (should not be exposed to the user then why its public? we have too many query APIs which should be private.

File: src/main/java/org/springframework/data/aerospike/core/AerospikeTemplate.java [hunk 2]
```diff
@@ -572,6 +583,53 @@ public <T, S> List<?> findByIdsInternal(Collection<?> ids, Class<T> entityClass,
         }
     }
 
+    public List<KeyRecord> findByIdsInternalWithoutMapping(Collection<?> ids, Class<?> entityClass,
```
Comment (roimenashe) [inline src/main/java/org/springframework/data/aerospike/core/AerospikeTemplate.java:586]: If its not part of AerospikeInternalOperations and is internal (should not be exposed to the user then why its public? we have too many query APIs which should be private.

File: src/main/java/org/springframework/data/aerospike/core/ReactiveAerospikeTemplate.java [hunk 3]
```diff
@@ -529,6 +532,36 @@ public <T, S> Flux<?> findByIdsInternal(Collection<?> ids, Class<T> entityClass,
             .map(keyRecord -> mapToEntity(keyRecord.key, target, keyRecord.record));
     }
 
+    public <T, S> Flux<KeyRecord> findByIdsInternalWithoutMapping(Collection<?> ids, Class<T> entityClass,
+                                                                  Class<S> targetClass,
+                                                                  Qualifier... qualifiers) {
+
```
Comment (roimenashe) [inline src/main/java/org/springframework/data/aerospike/core/ReactiveAerospikeTemplate.java:538]: redundant new line

File: src/main/java/org/springframework/data/aerospike/core/TemplateUtils.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import org.springframework.util.Assert;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public class TemplateUtils {
```
Comment (roimenashe) [inline src/main/java/org/springframework/data/aerospike/core/TemplateUtils.java:13]: Add lombok @UtilityClass annotation

File: src/main/java/org/springframework/data/aerospike/query/Qualifier.java [hunk 5]
```diff
@@ -390,4 +394,51 @@ protected void validate() {
...
+    public static Qualifier forIds(String... ids) {
+        return new Qualifier(new QualifierBuilder()
+            .setField(ID_VALUE)
+            .setFilterOperation(FilterOperation.EQ)
+            .setValue1(Value.get(Arrays.stream(ids).toList())));
+    }
+
+    /**
+     * Find id qualifier.
+     *
```
Comment (roimenashe) [inline src/main/java/org/springframework/data/aerospike/query/Qualifier.java:414]: add @param qualifiers to javadocs

File: src/main/java/org/springframework/data/aerospike/query/Qualifier.java [hunk 6]
```diff
@@ -390,4 +394,51 @@ protected void validate() {
             // do nothing
         }
     }
+
+    public static Qualifier forId(String id) {
```
Comment (roimenashe) [inline src/main/java/org/springframework/data/aerospike/query/Qualifier.java:398]: Add javadocs (forId and forIds)

File: src/test/java/org/springframework/data/aerospike/sample/PersonRepository.java [hunk 7]
```diff
@@ -124,6 +126,10 @@ public interface PersonRepository<P extends Person> extends AerospikeRepository<
 
     List<P> findByFirstNameAndLastName(String firstName, String lastName);
 
+    List<P> findByAgeBigInt(BigInteger age);
```
Comment (roimenashe) [inline src/test/java/org/springframework/data/aerospike/sample/PersonRepository.java:129]: I would change param name to `BigInteger` and method name to `findByAgeBigInteger` just to align with decimal and have right naming, `big int` can be mixed with long (for example in SQL Server `BIGINT` means long).

File: src/test/java/org/springframework/data/aerospike/sample/Person.java [hunk 8]
```diff
@@ -44,6 +46,8 @@ public class Person {
     private String firstName;
     private String lastName;
     private int age;
+    private BigInteger ageBigInt;
```
Comment (roimenashe) [inline src/test/java/org/springframework/data/aerospike/sample/Person.java:49]: `ageBigInteger` instead of `ageBigInt`.

**Review by roimenashe — APPROVED** [review]

**Review by reugn — APPROVED** [review]

**Pull request merged**